src/graph/optimizer/OptGroup.cpp

```diff
@@ -70,7 +80,8 @@ Status OptGroup::explore(const OptRule *rule) {
     NG_RETURN_IF_ERROR(groupNode->explore(rule));
 
     // Find more equivalents
-    auto status = rule->match(ctx_, groupNode);
+    std::vector<OptGroup *> boundary;
+    auto status = rule->match(ctx_, groupNode, boundary);
```

Comment:
Need not to collect the boundary of this rule. you could use the last MatchedResult to get the wanted OptGroup.

Reply:
Fixed.